gradle-suppressible-error-prone/src/main/java/com/palantir/gradle/suppressibleerrorprone/modes/interferences/RemoveUnusedAndApplyingInterference.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.suppressibleerrorprone.modes.interferences;
+
+import com.palantir.gradle.suppressibleerrorprone.modes.common.CommonOptions;
+import com.palantir.gradle.suppressibleerrorprone.modes.common.ModeName;
+import com.palantir.gradle.suppressibleerrorprone.modes.common.SpecificModeInterference;
+import java.util.Map;
+import java.util.Set;
+
+public final class RemoveUnusedAndApplyingInterference extends SpecificModeInterference {
```

Comment:
Isn't this identical to the suppressing + applying interference? What happens when we have suppressing + applying + removeUnused?

I'm honestly not convinced that this interference concept was necessary (mea culpa) - I think we may have been able to always use `-XepPatchChecks:` to enable patching on every bugchecker and always send a list of checks that need patching to `VisitorStateModifications`, which would remove fixes if it is not to be patched (when suppressing/applying/etc).

If we still need to keep interferences, I think we could use the `ModeInterference` interface directly, return all of suppress/apply/removeUnused as interfering if they have been enabled. The combine all of them together in `interfere` (and only have one interference between all three).